tests/chainer_tests/training_tests/extensions_tests/test_logreport.py

```diff
@@ -9,29 +11,32 @@
 
 class TestLogReport(unittest.TestCase):
 
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+
     def test_trigger(self):
-        trainer = _get_mocked_trainer((10, 'iteration'))
-        with mock.patch.object(extensions.LogReport, '_write') as mocked:
-            log_report = extensions.LogReport(trigger=(1, 'iteration'))
-            trainer.extend(log_report)
-            trainer.run()
-            mocked.assert_called()
-            self.assertEqual(mocked.call_count, 10)
-            self.assertEqual(len(log_report._log), 10)
+        trainer = _get_mocked_trainer(self.temp_dir, (10, 'iteration'))
+        log_report = extensions.LogReport(trigger=(1, 'iteration'))
+        trainer.extend(log_report)
+        trainer.run()
 
     def test_write_trigger(self):
-        trainer = _get_mocked_trainer((10, 'iteration'))
+        stops, aggregates, writes = 10, 2, 5
+        trainer = _get_mocked_trainer(self.temp_dir, (stops, 'iteration'))
         with mock.patch.object(extensions.LogReport, '_write') as mocked:
-            log_report = extensions.LogReport(trigger=(1, 'iteration'),
-                                              write_trigger=(10, 'iteration'))
+            log_report = extensions.LogReport(trigger=(aggregates, 'iteration'),
+                                              write_trigger=(writes, 'iteration'))
             trainer.extend(log_report)
             trainer.run()
             mocked.assert_called()
-            self.assertEqual(mocked.call_count, 1)
-            self.assertEqual(len(log_report._log), 10)
+            self.assertEqual(mocked.call_count, stops // writes)
+            self.assertEqual(len(log_report._log), stops // aggregates)
 
 
-def _get_mocked_trainer(stop_trigger=(10, 'iteration')):
+def _get_mocked_trainer(out, stop_trigger=(10, 'iteration')):
```

Comment:
It is easier to read to align the argument order to that of `Trainer` (i.e. swap `out` and `stop_trigger`. You can do it because the default argument of `stop_trigger` is not used).